### PR TITLE
fix: Viewer rendering for inconsistent object schemas

### PR DESCRIPTION
**Pull request by chrisgervang**
## Problem

When ViewerOp or handle preview tooltips receive arrays of objects with inconsistent schemas (different keys), the table rendering breaks or displays incorrect data. This happens when:
- Objects in an array have different properties
- Data comes from joins or merges with optional fields
- API responses have varying field structures

Visually, only the first row would be shown and the `<tr>`'s for subsequent rows would be empty.

## Solution

Add schema consistency check before rendering tables. If objects have different keys, fall back to JSON rendering instead.

## Changes

- Added `allHaveSameKeys` check in both ViewerOpComponent and HandlePreviewContent
- Compares sorted keys of all objects to detect schema inconsistencies
- Falls back to `ReactJson` component when schemas don't match
- Preserves table rendering for consistent schemas

## Testing

1. Create data with inconsistent object schemas:
   ```javascript
   [
     { name: "Alice", age: 30 },
     { name: "Bob", city: "NYC" },     // Different keys
     { name: "Charlie", age: 25, city: "SF" }  // Even more keys
   ]
   ```
2. View in ViewerOp - should render as JSON tree, not broken table
3. Hover over handle outputting this data - should render as collapsed JSON
4. Test with consistent schemas - should still render as table normally

## Edge Cases Handled

- Empty arrays
- Mixed object structures
- Null/undefined values
- Objects with function properties (filtered by isPlainObject)

### CODE REVIEW

**Review by akre54 — CHANGES_REQUESTED** [review]
I would rather we leave this one alone. I think its totally fine to have some empty cells  (though agree it would be nicer to have all the keys even if some cells are empty).

**Comment by chrisgervang** [comment]
Happy to explore different UI than falling back to JSON.

Please reconsider not fixing this issue though - I was convinced noodles was broken because the viewer didn't show all of the data it was given. I want to trust that it doesn't pick and choose which data to show, especially since I'm usually inspecting in uncertain circumstances ("is my flow working?" "Did I make a mistake somewhere?" "Why doesn't this output look right?")

**Comment by chrisgervang** [comment]
Added back table support for nonstandard arrays. Now every column renders and cells just have gaps when there isn't a value for the row

Before:
Screenshot 2026-01-20 at 7 06 22 PM

After:
Screenshot 2026-01-20 at 7 03 32 PM

File: noodles-editor/src/noodles/components/op-components.tsx [hunk 1]
```diff
@@ -357,7 +357,16 @@ function HandlePreviewContent({ data, name, type }: { data: unknown; name: strin
           isPlainObject(data[0]) &&
           Object.keys(data[0]).length < 10 ? (
           (() => {
-            const keys = Object.keys(data[0] || {})
+            // Derive union of all keys across rows to avoid silently dropping columns
+            const allKeys = new Set<string>()
+            for (const row of data) {
```
[the remark this answers is not in the document]
Comment (chrisgervang) [reply noodles-editor/src/noodles/components/op-components.tsx]: It only runs when there are less than 10 rows, so it shouldn't get too heavy
